source/common.ts

```diff
@@ -0,0 +1,479 @@
+export interface SetLocaleOptions extends LoadLocaleOptions {
+  noDownload?: boolean;
+}
+
+const i18n = {
+  _contextualLocale: new Meteor.EnvironmentVariable<string | undefined>(),
+  _deps: new Tracker.Dependency(),
+  _emitChange(locale?: string) {
+    i18n._events.emit('changeLocale', locale ?? i18n._locale);
+    i18n._deps.changed();
+  },
+  _events: new EventEmitter(),
+  _formatgetters: {
+    getJS: () => '',
+    getJSON: () => '',
+    getYML: () => '',
+  } as Pick<GetCacheEntry, 'getJS' | 'getJSON' | 'getYML'>,
+  _getConnectionId(connection?: Meteor.Connection | null) {
+    // Actual implementation is only on the server.
+    return undefined as string | undefined;
+  },
+  _getConnectionLocale(connection?: Meteor.Connection | null) {
+    // Actual implementation is only on the server.
+    return undefined as string | undefined;
+  },
+  _isLoaded: {} as Record<string, boolean>,
+  _loadLocaleWithAncestors(locale: string, options?: SetLocaleOptions) {
+    // Actual implementation is only on the client.
+    return Promise.resolve();
+  },
+  _locale: 'en',
+  _locales: LOCALES,
+  _logger(error: unknown) {
+    console.error(error);
+  },
+  _normalizeWithAncestors(locale = '') {
+    if (!(locale in i18n._normalizeWithAncestorsCache)) {
+      const locales: string[] = [];
+      const parts = locale.toLowerCase().split(/[\-_]/);
+      while (parts.length) {
+        const locale = parts.join('-');
+        if (locale in LOCALES) {
+          locales.push(LOCALES[locale][0]);
+        }
+
+        parts.pop();
+      }
+
+      i18n._normalizeWithAncestorsCache[locale] = locales;
+    }
+
+    return i18n._normalizeWithAncestorsCache[locale];
+  },
+  _normalizeWithAncestorsCache: {} as Record<string, readonly string[]>,
+  _translations: {} as JSONObject,
+  _ts: 0,
+  __(...args: unknown[]) {
+    // This will be aliased to i18n.getTranslation.
+    return '';
+  },
+  addTranslation(locale: string, ...args: unknown[]) {
+    // This will be aliased to i18n.addTranslations.
+    return {};
+  },
+  addTranslations(locale: string, ...args: unknown[]) {
+    const translation = args.pop() as JSONObject;
+    const path = args.join('.').replace(/(^\.)|(\.\.)|(\.$)/g, '');
+
+    if (typeof translation === 'string') {
+      set(i18n._translations, `${i18n.normalize(locale)}.${path}`, translation);
+    } else if (typeof translation === 'object' && !!translation) {
+      Object.keys(translation)
+        .sort()
+        .forEach(key => {
+          i18n.addTranslations(locale, `${path}.${key}`, translation[key]);
+        });
+    }
+
+    return i18n._translations;
+  },
+  createComponent(
+    translatorSeed?: string | ((...args: unknown[]) => string),
+    locale?: string,
+    reactjs?: typeof import('react'),
+    type?: React.ComponentType | string,
+  ) {
+    const translator = typeof translatorSeed === 'string'
+      ? i18n.createTranslator(translatorSeed, locale)
+      : typeof translatorSeed === 'undefined'
```

Comment:
Can't we just check for `undefined` explicitly? Since it is an argument there is no risk of this variable not being declared.
```suggestion
      : translatorSeed === undefined
```